### PR TITLE
Expand on how to use in a Dockerfile

### LINKED ISSUE #43: ENTRYPOINT vs CMD
Shouldn't `dumb-init` be used as `ENTRYPOINT` instead of `CMD`?

As I understand, default entrypoint for docker container is `/bin/sh -c`.
If you prefix your app `myapp` with `dumb-init` in `CMD`, what would be actually ran is `sh -c "dumb-init myapp"`, so PID1 would be still `sh -c`, not `dumb-init`.

### PR DESCRIPTION
**Pull request by chriskuehl**
Fixes #43

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -186,7 +186,21 @@ then just `pip install dumb-init`.
 ## Usage
 
 Once installed inside your Docker container, simply prefix your commands with
-`dumb-init`. For example:
+`dumb-init`.
+
+Within a Dockerfile, it's a good practice to use dumb-init as your container's
+entrypoint. An "entrypoint" is like a command that gets prepended to your `CMD`
```
Comment (asottile) [inline README.md]: s/like // imo maybe "partial command"?

**Comment by chriskuehl** [comment]
@asottile sgtm, look better?

File: README.md [hunk 2]
```diff
@@ -186,7 +186,21 @@ then just `pip install dumb-init`.
...
 Once installed inside your Docker container, simply prefix your commands with
-`dumb-init`. For example:
+`dumb-init`.
+
+Within a Dockerfile, it's a good practice to use dumb-init as your container's
+entrypoint. An "entrypoint" is a partial command that gets prepended to your
+`CMD` instruction, making it a great fit for dumb-init:
+
+```bash
+ENTRYPOINT ["/usr/bin/dumb-init", "--"]
```
Comment (asottile) [inline README.md]: shouldn't this be \`\`\`Dockerfile (also ps, you have no idea how long it took me to try and format this damn comment, and I still didn't get it code-formatted)

and perhaps: ```# Runs `/usr/bin/dumb-init -- /my/script --with --args` ```

Comment (chriskuehl) [inline README.md]: yeah, but last time I checked, GitHub didn't syntax highlight Dockerfile at all :\

comment sounds good

**Comment by asottile** [comment]
```Dockerfile
CMD foo
```

**Comment by asottile** [comment]
:shipit:

**Pull request merged**